source/fuzz/protobufs/spvtoolsfuzz.proto

```diff
@@ -907,6 +908,30 @@ message TransformationOutlineFunction {
 
 }
 
+message TransformationPermuteFunctionParameters {
+
+  // A transformation that, given a non-entry-point function taking n
+  // parameters and a permutation of the set [1, n]:
+  //   - Introduces a new function type that is the same as the original
+  //     function's type but with the order of arguments permuted
+  //     (only if it doesn't already exist)
+  //   - Changes the type of the function to this type
+  //   - Adjusts all calls to the function so that their arguments are permuted
+
+  // Function, whose parameters will be permuted
+  uint32 function_id = 1;
+
+  // An id for a new function type that might be created
+  uint32 fresh_type_id = 2;
+
+  // OpFunctionCall instructions that call our function
+  repeated InstructionDescriptor call_site = 3;
```

Comment:
I don't think these are needed as part of the transformation message.

If the function with id |function_id| exists, and if the permutation of function parameters is usable, then we can go ahead and find all of the calls to the function and permute their arguments.  We should not have to know about them in advance.

Reply:
Fixed in d7f8687